### DISCUSSION TITLE
services.netbox-housekeeping.depends_on contains an invalid type, it should be an array

### DISCUSSION THREAD

**Discussion by nncse** [netbox-community/netbox-docker#960] [General]
Hi, I tried to install netbox in docker using https://github.com/netbox-community/netbox-docker.git. But the install stops at docker-compose pull command because of the below error. Can somebody please share some guidance on how to resolve this issue? Thank you in advance.

root@ch1clwglmf002:/home/netbox-docker# docker-compose pull
ERROR: The Compose file './docker-compose.yml' is invalid because:
services.netbox-housekeeping.depends_on contains an invalid type, it should be an array
services.netbox-worker.depends_on contains an invalid type, it should be an array

**Comment by tobiasge** [comment]
You probably need to update your Docker compose.

**Reply by nncse** [reply]
docker compose version
Docker Compose version v2.16.0

**Reply by tobiasge** [reply]
That is good enough. But in the info you posted above you used `docker-compose` and not `docker compose`.

**Reply by nncse** [reply]
yes thank you @tobiasge. Now it is showing a different error.

Error response from daemon: driver failed programming external connectivity on endpoint netbox-docker-netbox-1 (981d257368cd90c59a54b02d7461caf592555cf01a1ad115a73ded6837cd6333): Bind for 0.0.0.0:8000 failed: port is already allocated

**Reply by nncse** [reply]
tcp        0      0 0.0.0.0:8000            0.0.0.0:*               LISTEN      1114/docker-proxy    off (0.00/0/0)
tcp6       0      0 :::8000                 :::*                    LISTEN      1119/docker-proxy    off (0.00/0/0)

**Reply by tobiasge** [reply]
> yes thank you @tobiasge. Now it is showing a different error.
> 
> Error response from daemon: driver failed programming external connectivity on endpoint netbox-docker-netbox-1 (981d257368cd90c59a54b02d7461caf592555cf01a1ad115a73ded6837cd6333): Bind for 0.0.0.0:8000 failed: port is already allocated

Something else is using this port. You need to set it to a free port in your `docker-compose.override.yml`.

**Reply by nncse** [reply]
Ok did this and reran docker compose up.

tee docker-compose.override.yml <<EOF
version: '3.4'
services:
  netbox:
    ports:
      - 8001:8080
EOF

**Reply by nncse** [reply]
Now, at the end it is saying it is unhealthy even though all checks are saying OK

WARN[0000] mount of type `volume` should not define `bind` option
[+] Running 3/4
 ⠿ Container netbox-docker-redis-cache-1  Running                                                      0.0s
[+] Running 6/6tbox-docker-postgres-1     Running                                                      0.0s
 ⠿ Container netbox-docker-redis-cache-1          Running                                              0.0s
 ⠿ Container netbox-docker-postgres-1             Running                                              0.0s
 ⠿ Container netbox-docker-redis-1                Running                                              0.0s
 ⠿ Container netbox-docker-netbox-1               Recreated                                            5.6s
 ⠿ Container netbox-docker-netbox-worker-1        Recreat...                                           2.1s
 ⠿ Container netbox-docker-netbox-housekeeping-1  R...                                                 2.1s
Attaching to netbox-docker-netbox-1, netbox-docker-netbox-housekeeping-1, netbox-docker-netbox-worker-1, netbox-docker-postgres-1, netbox-docker-redis-1, netbox-docker-redis-cache-1
netbox-docker-netbox-1               | ⚙️ Applying database migrations
netbox-docker-netbox-1               | 🧬 loaded config '/etc/netbox/config/configuration.py'
netbox-docker-netbox-1               | 🧬 loaded config '/etc/netbox/config/extra.py'
netbox-docker-netbox-1               | 🧬 loaded config '/etc/netbox/config/logging.py'
netbox-docker-netbox-1               | 🧬 loaded config '/etc/netbox/config/plugins.py'
netbox-docker-netbox-1               | Operations to perform:
netbox-docker-netbox-1               |   Apply all migrations: admin, auth, circuits, contenttypes, dcim, django_rq, extras, ipam, sessions, social_django, taggit, tenancy, users, virtualization, wireless
netbox-docker-netbox-1               | Running migrations:
netbox-docker-netbox-1               |   Applying dcim.0152_standardize_id_fields... OK
netbox-docker-netbox-1               |   Applying dcim.0153_created_datetimefield... OK
netbox-docker-netbox-1               |   Applying dcim.0154_half_height_rack_units... OK
netbox-docker-netbox-1               |   Applying dcim.0155_interface_poe_mode_type... OK
netbox-docker-netbox-1               |   Applying dcim.0156_location_status... OK
netbox-docker-netbox-1               |   Applying dcim.0157_new_cabling_models... OK
netbox-docker-netbox-1               |   Applying dcim.0158_populate_cable_terminations... OK
netbox-docker-netbox-1               |   Applying dcim.0159_populate_cable_paths... OK
netbox-docker-netbox-1               |   Applying extras.0069_custom_object_field... OK
netbox-docker-netbox-1               |   Applying extras.0070_customlink_enabled... OK
netbox-docker-netbox-1               |   Applying ipam.0055_servicetemplate... OK
netbox-docker-netbox-1               |   Applying ipam.0056_standardize_id_fields... OK
dependency failed to start: container netbox-docker-netbox-1 is unhealthy

**Reply by kkthxbye-code** [reply]
Override the healthcheck startup_time, the default is too low for initial migrations. Very common problem after it was added.

**Reply by nncse** [reply]
Thanks @kkthxbye-code, am a noob in docker, can you please provide command?

**Reply by aelmich** [reply]
@nncse I ran into the same. I edited docker-compose.yml 
adjust: 
healthcheck: 
start_period: 60s
to 
start_period: 90s
worked for me
